utils/nx-plugin/clone-build-targets-for-release.cjs

```diff
@@ -0,0 +1,130 @@
+    return dependency.startsWith('^');
+  }
+  return dependency.projects === 'dependencies';
+}
+
+/**
+ * @param {Readonly<Record<string, TargetConfiguration>>} projectTargets
+ * @param {string} targetName
+ * @returns {{target: TargetConfiguration, localDependencies: Record<string, TargetConfiguration>}}
+ */
+function cloneTarget(projectTargets, targetName) {
+  const dependsOn = projectTargets[targetName].dependsOn;
+  const dependenciesToClone = dependsOn
+    ? dependsOn.filter(
+        (targetDependency) => !isParentDependency(targetDependency)
+      )
+    : [];
+  /** @type {Record<string, TargetConfiguration>} */
+  const clonedDependencies = mergeObjects(
+    ...dependenciesToClone.map((targetDependency) => {
+      const targetDependencyName =
+        typeof targetDependency === 'string'
+          ? targetDependency
+          : targetDependency.target;
+      if (!(targetDependencyName in projectTargets)) {
+        return {};
+      }
+      const {target: clonedDependency, localDependencies: deepDependencies} =
+        cloneTarget(projectTargets, targetDependencyName);
+      return {...deepDependencies, [targetDependencyName]: clonedDependency};
+    })
+  );
+  return {
+    target: JSON.parse(JSON.stringify(projectTargets[targetName])),
+    localDependencies: clonedDependencies,
+  };
+}
+
+/**
+ * @param {TargetConfiguration} target
+ * @param {string} dependencyToReplace E.g.: `build` or `^build`
+ * @param {string[]} newDependencies
+ */
+function replaceDependency(target, dependencyToReplace, newDependencies) {
+  if (!target.dependsOn) {
+    return;
+  }
+  target.dependsOn = target.dependsOn.flatMap((dependency) => {
+    if (typeof dependency === 'string') {
+      if (dependency === dependencyToReplace) {
+        return newDependencies;
+      }
+      return [/** @type {string | TargetDependencyConfig} */ (dependency)];
+    }
+    const targetToReplace = dependencyToReplace.startsWith('^')
+      ? dependencyToReplace.slice(1)
+      : dependencyToReplace;
+    if (dependency.target === targetToReplace) {
+      return newDependencies;
+    }
+    return [/** @type {string | TargetDependencyConfig} */ (dependency)];
+  });
+}
+
+/**
+ * @param {Record<string, TargetConfiguration>} targets
+ * @param {string} oldName
+ * @param {string} newName
+ */
+function renameTarget(targets, oldName, newName) {
+  targets[newName] = targets[oldName];
+  delete targets[oldName];
+  for (const target of Object.values(targets)) {
+    // We only cloned dependencies which are defined locally. We don't need to replace parent dependencies.
+    replaceDependency(target, oldName, [newName]);
+  }
+}
+
+/**
+ * @param {ProjectGraphProjectNode} project
+ */
+function addReleaseBuildTargets(project) {
+  const projectTargets = project.data.targets || {};
+  if (!('build' in projectTargets)) {
+    return;
+  }
+  const {target: buildTarget, localDependencies: buildTargetDependencies} =
+    cloneTarget(projectTargets, 'build');
```

Comment:
We want to clone both the `build` target itself, and all the **local** (non-parent) targets it depends on.

To understand this, take a look at `atomic`'s targets:
```json
{
  "targets": {
    "cached:build:stencil": {
      "dependsOn": ["^build", "build:locales"],
      "inputs": ["^production", "production"],
      "outputs": [
        "..."
      ],
      "executor": "...",
      "options": {
        "..."
      }
    },
    "validate:definitions": {
      "dependsOn": ["^build", "cached:build:stencil"],
      "inputs": [],
      "outputs": [],
      "executor": "...",
      "options": {
        "..."
      }
    },
    "build": {
      "dependsOn": ["cached:build:stencil", "validate:definitions"],
      "inputs": [],
      "outputs": [],
      "executor": "nx:noop"
    }
  }
}
```

As you can see, if we were to just clone the `build` target, it would queue  `cached:build:stencil` and `validate:definitions`. Subsequently, these targets would queue up `^build` and `build:locales`.

Queueing up `^build` is problematic. It will cause parent projects to be rebuilt by executing their `build` target, which won't wait for the version to be bumped. This could cause Headless to be built before its version is bumped, which would cause Atomic to use an old build of Headless, which contains the wrong Headless `engine.state.version`.